From 710003dd542d85052aa9f3d08e8a0cf9daa43949 Mon Sep 17 00:00:00 2001
From: Jiaming Huang <jiaming.huang@amlogic.com>
Date: Thu, 9 Jul 2015 09:36:03 +0800
Subject: [PATCH 5698/5965] PD#108932: g9tv: lcd: close lcd phy function when
 suspend

This changing close lcd phy to close the lcd signal
when suspend.

Validated on N361.

Change-Id: I1dce9f74fd4840576bfdc404daec714505e7c0e1
---
 drivers/amlogic/display/lcd/aml_tv_lcd.c                 | 3 ++-
 drivers/amlogic/display/lcd/aml_tv_lcd_port/lcd_common.c | 4 ++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/display/lcd/aml_tv_lcd.c b/drivers/amlogic/display/lcd/aml_tv_lcd.c
index 76f1abee8f6f..ed83b00b570b 100644
--- a/drivers/amlogic/display/lcd/aml_tv_lcd.c
+++ b/drivers/amlogic/display/lcd/aml_tv_lcd.c
@@ -343,10 +343,11 @@ static void panel_power_ctrl(Bool_t status)
 		mdelay(pConf->lcd_power_ctrl.panel_power->panel_on_delay);
 
 	} else {
+		mdelay(pConf->lcd_power_ctrl.panel_power->panel_off_delay);
 		amlogic_gpio_request(pConf->lcd_power_ctrl.panel_power->gpio,owner);
 		amlogic_gpio_direction_output(pConf->lcd_power_ctrl.panel_power->gpio,
 									  pConf->lcd_power_ctrl.panel_power->off_value,owner);
-		mdelay(pConf->lcd_power_ctrl.panel_power->panel_off_delay);
+
 	}
 }
 
diff --git a/drivers/amlogic/display/lcd/aml_tv_lcd_port/lcd_common.c b/drivers/amlogic/display/lcd/aml_tv_lcd_port/lcd_common.c
index b5a7bbdb4268..bb5ce87d3658 100644
--- a/drivers/amlogic/display/lcd/aml_tv_lcd_port/lcd_common.c
+++ b/drivers/amlogic/display/lcd/aml_tv_lcd_port/lcd_common.c
@@ -190,6 +190,10 @@ void _disable_display_driver(void)
 	else
 		aml_set_reg32_bits(P_HHI_VID_CLK_CNTL, 0, 0, 5);		//close vclk1 gate: 0x105f[4:0]
 
+	aml_write_reg32(P_HHI_DIF_CSI_PHY_CNTL1, 0x0);   //close phy
+	aml_write_reg32(P_HHI_DIF_CSI_PHY_CNTL2, 0x0);
+	aml_write_reg32(P_HHI_DIF_CSI_PHY_CNTL3, 0x0);
+
 	TV_LCD_INFO("disable lcd display driver.\n");
 }
 
-- 
2.19.0

